### ISSUE TITLE
Git repo integration

### ISSUE THREAD

**Issue by stone212** [nextcloud/deck#849]
**Describe the solution you'd like**

It would be helpful (and a GitLab replacement in our case) if we could associate each board with a Git repository.

**Describe alternatives you've considered**
Two ways this could work:

1. You set the parent repo and credentials in the board settings and each new board creates a new repo under it.

2. (better) you set the repo URL and scheme in the settings for each kanban board.  

3. Someday a Git plugin for Nextcloud could integrate directly.

**Additional context**

Bosses can't use GitLab for some reason and for some reason they love Nextcloud.  So they send text messages to you that link to a LibreOffice doc in Nextcloud that just has a task that relates to a Gitlab project you are already working on.

**Comment by juliushaertl** [comment]
As this sounds like a nice feature, the requests for this are quite low. Currently there are no plans to implement such a feature. Thus I will close this ticket for now. This does not mean we don't want this feature, but it is simply not on our roadmap for the near future. If somebody wants to implement this feature nevertheless we are happy to assist and help out. However it would require some deeper planning upfront, so we make sure we create a proper concept of how repositories and issues are integrated.

In general I think solutions that come with those code hosting plattforms like projects in github/gitlab are far more suitable for those tasks, since they are integrated right into them and do not add another layer of software to the process.

If you wish to have this feature implemented by the Nextcloud GmbH there is the option for consulting work on top of your Nextcloud Enterprise subscription to get your features implemented.

**Comment by stone212** [comment]
@juliushaertl 

> In general I think solutions that come with those code hosting plattforms like projects in github/gitlab are far more suitable for those tasks, since they are integrated right into them and do not add another layer of software to the process.

Well is there a way to integrate a GL/GH account into NC?  The use case is stupid managers who love NC but don't log into GL/GH.

Anyway that method defeats the purpose of NC which is to control your own data.

**Comment by stone212** [comment]
@juliushaertl That was funny but it does not also speak to the problem of requiring two systems.  So you prefer to be funny than to have an actual discussion about improving the product.  Nextcloud people are really the worst.  Every other project the devs care about making a good product, not just a popular one.  Nextcloud no.

**Comment by jancborchardt** [comment]
@stone212 I would like to point out that we are a community project and want to keep this a place that is nice and positive for everyone. These kinds of comments are not welcome. Please try to be more respectful going forward, see our Code of Conduct for the why and what: https://nextcloud.com/code-of-conduct/

**Comment by stone212** [comment]
@jancborchardt 

What sort of comments?  Critical ones?  Well that's too bad.  And your code of conduct does not apply on a 3rd party site like GitHub any more than my personal code of conduct applies to you.  Really, please don't be so silly.

**Comment by jancborchardt** [comment]
@stone212

> What sort of comments? Critical ones? Well that's too bad.

You saying:

> Nextcloud people are really the worst.

is inconsiderate and disrespectful. Both are points in our Code of Conduct: https://nextcloud.com/code-of-conduct/


---------------------

> And your code of conduct does not apply on a 3rd party site like GitHub any more than my personal code of conduct applies to you. Really, please don't be so silly.

As I already said in the other issue where you decided to insult people at https://github.com/nextcloud/documentation/issues/1142:

> **Our code of conduct applies in all of our online and offline spaces.** This includes all our repositories on Github and our forum too. And your tone is not welcome in our community.

**Comment by shironextcloud** [comment]
@stone212 
Whether you want to acknowledge it or not, our Code of Conduct is the rules we decided everyone has to respect in order to be welcome in our community. This includes, of course, our organization on Github. The rules are very simple, all we ask you to do is to be respectful when you interact with people here. 

Unlike you, Github acknowledges that every organization has the right to define guidelines for their community. Look, they even encourage communities to do so, by providing a help page on the topic: https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/

In short, if you refuse to be respectful in your interactions in the Nextcloud organization on Github, we can block you from commenting here. Your behavior has been now reported several times to us, and as much as we're all about giving people one more chance, we're getting tired of seeing you wearing out members of our community. Consider this as your last warning - one more complaint about you, and we will have no other option than blocking you.

Make the best of this additional chance we are giving you to show that you can be a nice person :)
